### ISSUE TITLE
After adding the CustomTooltip feature, cells that contain html tags with some attributes are being excluded

### ISSUE THREAD

**Issue by mcallegario** [ghiscoding/slickgrid-universal#548]
### Your Environment

| Software               | Version(s) |
| --------------------- | ---------- |
| Slickgrid-Universal  | v0.19.0 |
| TypeScript               | x.y        |
| Package Name        | x.y        |

### Describe the Bug

After adding the CustomTooltip feature, cells that contain html tags with some attributes, for example "onclick", are being excluded as shown in the images below.

### Steps to Reproduce

fills cell content with html tag with "onclick" attribute, example:

`<span class="mdi mdi-check-circle-outline" onclick="alert('Show Hello world!')" title="Hello world!">Hello</span> Task 1`

### Expected Behavior

```
`<div class="slick-cell l1 r1 true">
   <div class="editing-field">
      <span class="mdi mdi-check-circle-outline" onclick="alert('Show Hello world!')" title="Hello world!">Hello</span> Task 1
   </div>
</div>`
```

### Current Behavior

![image](https://user-images.githubusercontent.com/44271732/141888201-7d152ca3-bf0c-4094-a696-76d0df9a7ba1.png)

### Possible Solution

I do not know  :-(

### Code Sample

![image](https://user-images.githubusercontent.com/44271732/141888429-410c98b7-dc4e-482f-9ff8-ee25074c0a32.png)

**Comment by ghiscoding** [comment]
that is to be expected, you are never ever supposed to add JavaScript code into your code because that is exactly how XSS Scripting Attacks occur... so please never do it that way and yes script tags are now completely removed from the output to block these kind of attacks in 2 different ways
1. recently created this SlickGrid [PR #652](https://github.com/6pac/SlickGrid/pull/652) in the core lib (uses simple and basic RegEx, removes basic code like JavaScript and other tags). No external lib
2.  on the other end Slickgrid-Universal uses [DOMPurify](https://github.com/cure53/DOMPurify) in code like Filters, Editors and Export Services and Custom Tooltips, whatever else is are done via number 1.

So to go back on the `onClick`, like I said never do it that way, instead you can do it in 2 ways
1. `onClick` SlickGrid event triggered by the entire grid (anytime a cell is clicked)
2. `onCellClick` event defined on a Column Definition
   - it's actually a utility wrapper, it uses the `onClick` but I just added code in the lib to make it easy to hook it directly to the column

I prefer to use number 2 most of the time, for example this:
https://github.com/ghiscoding/slickgrid-universal/blob/073f79eabe7762c02c15fd79b603bf7bcc4e2f71/examples/webpack-demo-vanilla-bundle/src/examples/example11.ts#L258-L273


so long story short, there is nothing to do on my side but you'll have to fix your side

**Comment by mcallegario** [comment]
Sorry for my lack of knowledge and
again, thank you very much for your explanation.

**Comment by ghiscoding** [comment]
no problem, stay safe in every meaningful ways 😉 
Also consider asking questions on Stack Overflow, I answer few questions there too and it's a better tool for questions in general. 

Cheers

**Comment by ghiscoding** [comment]
hey I was looking at your code and you mentioned that the `title` got removed as well but I see you're using the Column Definition `title` property, however in my new Example 16 I used the `name` property and the tooltip was working fine. I don't know what the `title` is for (in fact I never use it), it might be for tooltip but I'm not sure, I typically always use these 3 properties (`id`, `name` and `field`). Anyway I just thought of letting you know, also there's a new Example 16 and a new [Custom Tooltip - Wiki](https://github.com/ghiscoding/slickgrid-universal/wiki/Custom-Tooltip-(plugin)) as well for that new tooltip plugin I created 😃 Hope it's useful, cheers

https://github.com/ghiscoding/slickgrid-universal/blob/9c70be5e3633949937cc760d0f5c9319e36cd92a/examples/webpack-demo-vanilla-bundle/src/examples/example16.ts#L93

**Comment by mcallegario** [comment]
@ghiscoding 

Understood, but my biggest problem currently is the exclusion of 'onclick' because it has in several places the html tag with this attribute defined, which is making the switch to a newer version of slickgrid unfeasible.
By any chance, would it have a parameter to temporarily disable XSS?

**Comment by ghiscoding** [comment]
Hmm you could maybe try to fix the SlickGrid (core) version to the previous one (before PR [#652](https://github.com/6pac/SlickGrid/pull/652) got merged) with `"slickgrid": "^2.4.42"` but that might get overridden by Slickgrid-Universal itself because it requires latest SlickGrid version and that was releases at the same time as the new Slickgrid-Universal Custom Tooltip feature. In Slickgrid-Universal, there's the `sanitizer` that callback that you can override

https://github.com/ghiscoding/slickgrid-universal/blob/d15022b14b8f93b84e2e1d89a42d79fc16c40c2b/packages/common/src/interfaces/gridOption.interface.ts#L553-L557

but I think that in your code it's most probably the sanitizer that was added by PR [#652](https://github.com/6pac/SlickGrid/pull/652) which is removing the onclick event and that sanitizer doesn't have a setter, but it's JS so who knows maybe you can override directly, it's a different method name in SlickGrid core [sanitizeHtmlString](https://github.com/6pac/SlickGrid/blob/master/slick.grid.js#L6123).
So things you could try are
1. override Slickgrid-Universal `sanitizer` in your grid options
   - `this.gridOptions = { sanitizer: (dirtyHtml: string) => dirtyHtml , ... /* more options */ }`
2. try overriding SlickGrid (core) `sanitizeHtmlString` that is coming directly from SlickGrid object (though I'm not sure if that would work, give it a try)
   - `(this.sgb.slickGrid as any).sanitizeHtmlString = (dirtyHtml: string) => dirtyHtml` 
3. OR revert to previous SlickGrid version "slickgrid": "^2.4.42" (you can try adding it to your `package.json`, though I suspect Slickgrid-Universal might still end up using latest SlickGrid version anyway)
4. OR revert to previous Slickgrid-Universal version but you'll miss out on the new Custom Tooltip feature.
5. there's maybe a last option but it's less feasible and would take more time, which would be to make the `sanitizeHtmlString` a setter the same way as Slickgrid-Universal has so that you could change it, however that would require 2 version releases (1 release on SlickGrid core, which I'm not the owner, and 1 release of Slickgrid-Universal) and it's even less feasible now because I'm preparing an official Slickgrid-Universal 1.0.0 release in the next couple weeks which will include another breaking change if you're using SASS variables from the lib (all SASS variables will require in the future the `$slick-` prefix as per this new PR #556)

I would try number 1 & 2 first to see if that works, crossing my fingers and hoping that would work 🤞 

#### Conclusion
I understand that it might be some time to refactor all of that but it was really changed for security reasons, because having JS code inside a formatter is really considered a security hole and was patched for that reason. It also wasn't released as a breaking change version because we shouldn't use JS code in Formatters in the first place and I assumed wrongly that everyone knew about this, but now I know that we might need Wiki docs updated to make it clear on how to use click and other events the correct and secure way (I'm just not sure in which Wiki I should add this though). 

Let me know if that worked, if not we can try a few other things

**Comment by mcallegario** [comment]
@ghiscoding 

I managed to solve this by implementing a new option in slickGrid and manually updated it in the slickGrid-Universal project.

Changes SlickGrid:
![image](https://user-images.githubusercontent.com/44271732/143665859-724c5076-ca32-426c-b303-c3c8161fb782.png)


Changes SlickGrid-Universal:
![image](https://user-images.githubusercontent.com/44271732/143665878-5c8983f3-0722-413e-8cd2-5d2cdc04a618.png)

**Comment by ghiscoding** [comment]
@mcallegario that is just a temporary code change for your local development is it? 

Because I would rather do a different implementation, and I will probably do a PR for this in the SlickGrid core lib. The code would allow to provide a `sanitizer` grid option, which Slickgrid-Universal already have in place in grid option interface [here](https://github.com/ghiscoding/slickgrid-universal/blob/master/packages/common/src/interfaces/gridOption.interface.ts#L557).

```js
function sanitizeHtmlString(dirtyHtml) {
  var sanitizer = options.sanitizer || function (dirtyHtml) {
    return dirtyHtml.replace(/(\b)(on\S+)(\s*)=|javascript:([^>]*)[^>]*|(<\s*)(\/*)script([<>]*).*(<\s*)(\/*)script(>*)|(&lt;)(\/*)(script|script defer)(.*)(&gt;|&gt;">)/gi, '');
  }
  return typeof dirtyHtml === 'string' ? sanitizer(dirtyHtml) : dirtyHtml;
}
```
so adding this to SlickGrid lib would make it work for both libs all at once by simply doing something like this
```ts
this.gridOptions = {
  sanitizer: (dirtyHtml) => dirtyHtml, // return the string without sanitizing it
}
```

here's a code test in JSBin

![image](https://user-images.githubusercontent.com/643976/143667226-1e4bd4a2-63df-48d5-81e8-b407c98bf1bf.png)

Lastly, I'm actually using this `sanitizer` option in our Salesforce implementation because original sanitizer which uses [DOMPurify](https://github.com/cure53/DOMPurify) throws some errors in Salesforce. So instead of using DOMPurify, I use the same function I implemented in SlickGrid core lib.

https://github.com/ghiscoding/slickgrid-universal/blob/43673f4408b1730aed9ef8e66c30abf49c829832/packages/vanilla-bundle/src/salesforce-global-grid-options.ts#L58

**Comment by mcallegario** [comment]
@ghiscoding 
Exactly, I made these changes only in my local code to check if it would work.
And then exchange that idea with you.
I understood your suggestion and I also thought it was better than mine.
Thank you one more time.
